crates/madara/client/settlement_client/src/client.rs

```diff
@@ -0,0 +1,87 @@
+use crate::gas_price::L1BlockMetrics;
+use crate::messaging::CommonMessagingEventData;
+use crate::state_update::StateUpdate;
+use async_trait::async_trait;
+use futures::Stream;
+use mc_db::l1_db::LastSyncedEventBlock;
+use mc_db::MadaraBackend;
+use mp_utils::service::ServiceContext;
+use starknet_types_core::felt::Felt;
+use std::sync::Arc;
+
+pub enum ClientType {
+    ETH,
+    STARKNET,
+}
+
+#[async_trait]
+pub trait ClientTrait: Send + Sync {
+    // Configuration type used for initialization
+    type Config;
+
+    // Get client type
+    fn get_client_type(&self) -> ClientType;
+
+    // Basic getter functions
+    fn get_l1_block_metrics(&self) -> &L1BlockMetrics;
+
+    // Create a new instance of the client
+    async fn new(config: Self::Config) -> anyhow::Result<Self>
+    where
+        Self: Sized;
+
+    // Get the latest block number
+    async fn get_latest_block_number(&self) -> anyhow::Result<u64>;
+
+    // Get the block number of the last occurrence of a specific event
```

Comment:
let's change the comment, given it's not generic now
```suggestion
    // Get the block number of the last occurrence of the state update event
```

Reply:
Done